### ISSUE TITLE
bot immortal and not moving

### ISSUE THREAD

**Issue by DhyanCanPlay** [mhsjlw/pocket-minecraft-protocol#523]
hey so idk why but my bot became immortal (not on creative) and doesnot perform movement. before update it was great but now it has problems. i dont have issue in imortality but i need it to move to location

**Comment by extremeheat** [comment]
Please use the Discord for support questions. If you have an issue please provide full details including:
* your code
* the expected result
* the observed result

If you don't provide steps to reproduce issues, then it can't be fixed

**Comment by DhyanCanPlay** [comment]
no one answered in discord
so i came here

**Comment by afarhansib** [comment]
i have this problem too

i create a normal client, login to server

- EXPECTED 
the bot just join as survival player, holding sword, can be killed

- OBSERVED
the bot join, but floating in air when i break the block beneath, it cannot be hit

**Comment by fancode343** [comment]
Same with me

There is a code to move the bot but its so complicated

**Comment by afarhansib** [comment]
> Same with me
> 
> There is a code (packet) to move the bot but its so complicated

but does it work? does the bot actually moves?

**Comment by Slauh** [comment]
> but does it work? does the bot actually moves?

yes it does but like he said it is complicated and you have to send it a lot the packet name is something along the lines of "player_auth_movement" in this it a ton a arguments you can get the data from a relay there might be two packets that move the player but this is one

**Comment by Slauh** [comment]
you also have to send data to get it out of immortal state not sure what it is atm

Sending the movement packets should fix the floating issue not too sure though

**Comment by kwoowk** [comment]
```js
client.on("play_status", (packet) => {
     if (packet.status == "player_spawn") {
          client.write("serverbound_loading_screen", {
               type: 2
          });
     };
});
```

This should be able to fix the immortal problem. I found this in the prismarine discord.

**Comment by Slauh** [comment]
> ```js
> client.on("play_status", (packet) => {
>      if (packet.status == "player_spawn") {
>           client.write("serverbound_loading_screen", {
>                type: 2
>           });
>      };
> });
> ```
> 
> This should be able to fix the immortal problem. I found this in the prismarine discord.

I tried this before today and it didn't seem to work at least on my side

**Comment by DhyanCanPlay** [comment]
Any solution to move the bot?? like I tried all thing and it doesn't work. Please if anyone can help me.

**Comment by TSL534** [comment]
Try to use the player_auth_input Packet (Everything over 5 Blocks of the Bot Position is getting blocked and make sure it moves with the Ticks)

**Comment by Slauh** [comment]
> Try to use the player_auth_input Packet (Everything over 5 Blocks of the Bot Position is getting blocked and make sure it moves with the Ticks)

you have to send more than that to get it to not be immortal 
```js
let runtimeEntityId;

client.on('start_game', (packet) => {
  runtimeEntityId = packet.runtime_entity_id
  console.log(`Runtime Entity ID saved: ${runtimeEntityId}`);
});



client.on('start_game', (packet) => {
client.queue('serverbound_loading_screen', {
  "type": 1
})
client.queue('serverbound_loading_screen', {
  "type": 2
})
client.queue('interact', {
  "action_id": "mouse_over_entity",
  "target_entity_id": 0n,
  "position": {
    "x": 0,
    "y": 0,
    "z": 0
  }
})
client.queue('set_local_player_as_initialized', {
  "runtime_entity_id": `${runtimeEntityId}`
})
})
```
and alongside send the movement packet

**Comment by afarhansib** [comment]
> > Try to use the player_auth_input Packet (Everything over 5 Blocks of the Bot Position is getting blocked and make sure it moves with the Ticks)
> 
> you have to send more than that to get it to not be immortal
> 
> let runtimeEntityId;
> 
> client.on('start_game', (packet) => {
>   runtimeEntityId = packet.runtime_entity_id
>   console.log(`Runtime Entity ID saved: ${runtimeEntityId}`);
> });
> 
> 
> 
> client.on('start_game', (packet) => {
> client.queue('serverbound_loading_screen', {
>   "type": 1
> })
> client.queue('serverbound_loading_screen', {
>   "type": 2
> })
> client.queue('interact', {
>   "action_id": "mouse_over_entity",
>   "target_entity_id": 0n,
>   "position": {
>     "x": 0,
>     "y": 0,
>     "z": 0
>   }
> })
> client.queue('set_local_player_as_initialized', {
>   "runtime_entity_id": `${runtimeEntityId}`
> })
> })
> and alongside send the movement packet

just confirming i have tested this on 1.21.51 and it works!

the bot now can be in survival and got hurt. just not updating its position with knockback or pushed by water or breaking its platform.

my purpose is get the bot to throw a trident (can using regular client) and afk for a farms. previously when the bot immortal, the trident cannot hit entity, now it can!
